addons/l10n_in_ewaybill_irn/models/l10n_in_ewaybill.py

```diff
@@ -23,6 +23,7 @@ def _compute_is_process_through_irn(self):
                 # E-way Bill is not generated by IRN for document types of Debit Note and Credit Note.
                 and not (account_move_id.move_type == 'out_refund' or account_move_id.debit_origin_id)
                 and account_move_id._l10n_in_check_einvoice_eligible()
+                and account_move_id.l10n_in_edi_status == 'sent'
```

Comment:
```suggestion
                and account_move_id.l10n_in_edi_status == 'sent'
```
If it's an E-invoiced move then there is no need to check `_l10n_in_check_einvoice_eligible`. Better update `api.depends` too